pkg/image/trigger/buildconfigs/buildconfigs.go

```diff
@@ -175,8 +176,12 @@ func (r *buildConfigReactor) ImageChanged(obj runtime.Object, tagRetriever trigg
 			newSource = true
 		}
 
-		// LastTriggeredImageID is an image ref, despite the name
-		if latest == p.LastTriggeredImageID {
+		// Determine whether this trigger has fired previously and is recorded in status;
+		// openshift-apiserver updates the status of image change triggers,
+		// including resolving LastTriggeredImageID and LastTriggerTime.
+		pStatus := ocmbuildutil.GetImageChangeTriggerStatusForImageChangeTrigger(p, bc)
+		if pStatus != nil && latest == pStatus.LastTriggeredImageID {
```

Comment:
this is back to what we want ... where we only need to check our new status last triggered image ID to make sure we do not need to trigger a build